TEMPLATES/mega-linter.yml

```diff
@@ -40,129 +120,227 @@ jobs:
     name: MegaLinter
     runs-on: ubuntu-latest
 
-    # Give the default GITHUB_TOKEN write permission to commit and push, comment
-    # issues & post new PR; remove the ones you do not need
+    ##
+    # This job's `GITHUB_TOKEN` or your added `PAT` must have the following
+    # permissions enabled to `write` for MegaLinter to function.
+    #
+    # Always aim to provide as few permissions as possible for personal access
+    # tokens. By default, all permissions are set to read only. Write
+    # permissions must be set for the following scopes in the described
+    # circumstances:
+    #
+    #   To have MegaLinter write and update a linting report comment on Pull
+    #   Requests, the `pull-requests` and `issues` permissions must be `write`.
+    #
+    #   To enable `APPLY_FIXES`, the `pull-requests` and `contents` permissions
+    #   must be `write`.
+    #
+    # More info at:
+    # https://docs.github.com/en/actions/security-guides/automatic-token-authentication
+    #
     permissions:
       contents: write
       issues: write
       pull-requests: write
 
     steps:
 
-      # Git Checkout
+      ##
+      # Clone the Repository for linting.
+      #
+      # More info at:
+      # https://github.com/actions/checkout
+      #
       - name: Checkout Code
         uses: actions/checkout@v4
         with:
           token: ${{ secrets.PAT || secrets.GITHUB_TOKEN }}
 
-          # If you use VALIDATE_ALL_CODEBASE = true, you can remove this line to
-          # improve performance
+          ##
+          # A fetch depth of `0` pulls the entire Repository's history,
+          # Branches, and Tags. Limiting this to `1` (or commenting out the
+          # line, as this is the default) increases efficiency, but is only
+          # applicable when `VALIDATE_ALL_CODE_BASE = true`.
+          #
           fetch-depth: 0
 
-      # MegaLinter
+      ##
+      # Run MegaLinter.
+      #
+      # More info at:
+      # https://github.com/oxsecurity/megalinter
+      #
       - name: MegaLinter
+        id: ml
 
-        # You can override MegaLinter flavor used to have faster performances
-        # More info at https://megalinter.io/flavors/
+        ##
+        # The full MegaLinter project can be over 8GB in size! This is due to it
+        # including every linter available for the full MegaLinter project. It
+        # is highly recommended to select a more refined "flavor" that better
+        # matches your project by changing the image used here.
```

Comment:
Running MegaLinter (could) suggest a flavor that matches the file types in the project, it would be noted in the beginning of the action's output (log).

Reply:
I usually enable that feature by default on my projects, in case I miss something as MegaLinter evolves.

Is your recommendation here to inform the user as-such here in the documentation? Or something else? Let me know.

Reply:
Rereading my answer and yours I realize I wasn't really clear, and looks like I didn't know what I wanted. 

Flavor suggestions normally happen when running. 
https://github.com/oxsecurity/megalinter/blob/3d17bf98d60e9d2b546df1e3792da7e4d372d6ab/megalinter/MegaLinter.py#L260-L269


It is a really good idea to select a more adapted flavor, but the user should know that they aren't expected to already know it before starting to use Megalinter (or when configurating it). Megalinter will tell them if a flavor adapted for their project exists, and that they can place it there.

Reply:
Oh yeah, I'm understanding you - I use that feature regularly, and keep it on so I get alerted if better flavors get released :)

I guess my question is what you want to do with the documentation here. In my opinion, repeating that selecting a flavor has benefits isn't a bad thing. Maybe adding a line something like this?

```
# The full MegaLinter project can be over 8GB in size! This is due to it
# including every linter available for the full MegaLinter project. It
# is highly recommended to select a more refined "flavor" that better-
# matches your project by changing the image used here.
#
# For example, a Python project may elect to use:
#
#   oxsecurity/megalinter/flavors/python@<%= GITHUB_ACTION_VERSION %>
#
# Don't worry, however. MegaLinter is intelligent and will scan your
# repository the first time you run it and make recommendations as to
# which flavor best-suits your codebase. We recommend performing a
# first-run on your project as this is configured, and reading the logs
# carefully to see if MegaLinter recommends a specific flavor for your
# project setup near the beginning of the output.
#
# More info at:
# https://megalinter.io/flavors/
```

Reply:
Since it isn't only for the first run, and that the note is at the end of the run (beneath the summary table when ran in a console locally), I would think about something like


```yaml
# The full MegaLinter project can be over 8GB in size! This is due to it
# including every linter available for the full MegaLinter project. It
# is highly recommended to select a more refined "flavor" that better-
# matches your project by changing the image used here.
#
# For example, a Python project may elect to use:
#
#   oxsecurity/megalinter/flavors/python@<%= GITHUB_ACTION_VERSION %>
#
# Don't worry, however. MegaLinter is intelligent and will scan your
# repository and make recommendations as to
# which flavor best-suits your codebase. We recommend performing a
# first-run on your project as is already configured, below.
# If MegaLinter recommends a
# specific flavor for your project setup, it will be shown after the summary table at the end of the run.
#
# More info at:
# https://megalinter.io/latest/flavors/
```

I'm writing on my cellphone, so the code block probably isn't wrapped correctly for line lengths

Reply:
Haha, yeah it kept going, but I like the add. I'll fix that with the outstanding spelling needs and mark this resolved.